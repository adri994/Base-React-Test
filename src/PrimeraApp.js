import React from 'react'
import PropTypes from 'prop-types'


const PrimeraApp = ({saludo,subtitulo}) =>{

    return(
        <>
            <h1>{saludo}</h1>
            <p>{subtitulo}</p>
        </>
    ); 

}

//Indicar que se va a recibir
PrimeraApp.propTypes ={
    saludo:PropTypes.string.isRequired
}

//Otra forma de poner parametros por defectos

PrimeraApp.defaultProps = {
    subtitulo:"vista"
}

export default PrimeraApp;