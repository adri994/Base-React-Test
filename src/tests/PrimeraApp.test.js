//import { render } from "@testing-library/react"
import PrimeraApp from "../PrimeraApp"
import React from 'react'
import { shallow } from 'enzyme'
import "@testing-library/jest-dom"


describe('Prueba en PrimeraApp', () => {
    
/*     test('Debe mostrar hola mundo', () => {

        const saludo ="hola mundo"

        //Sirve para renderizar
        const { getByText } = render(<PrimeraApp saludo={saludo} />)

        //Par que funcione este archivo tenemos que crear un archivo llamado setupTest
        //Noormalemnte cuando creamos un nuevo proyeto aparece
        expect( getByText(saludo) ).toBeInTheDocument()
        
    })
     */

    test('Debe mostrar <PrimeraApp /> correctamente', () => {
        
        //renderizar
        const wrapper =shallow( <PrimeraApp saludo="hola mundo" />)
        //Tenemso que descargar otra aplicacion
        expect( wrapper ).toMatchSnapshot();
    })

    test('debe mostrar el subtitulo enviado por props', () => {
        
        const subtitulo = "hola sammy"
        const wrapper =shallow( 
            <PrimeraApp 
                saludo="hola mundo" 
                subtitulo={subtitulo}/>)

        //Se comporta como el querySelector
        const textParrafo = wrapper.find("p").text()
        
        expect(textParrafo).toBe(subtitulo)
    })
    
    
})
