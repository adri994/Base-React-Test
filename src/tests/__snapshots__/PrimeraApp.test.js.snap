// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en PrimeraApp Debe mostrar <PrimeraApp /> correctamente 1`] = `
<Fragment>
  <h1>
    hola mundo
  </h1>
  <p>
    vista
  </p>
</Fragment>
`;
